contract-message-test/tests/harness.rs

```diff
@@ -1,184 +1,672 @@
+
+        // Verify the message value was received by the test contract
+        let provider = wallet.get_provider().unwrap();
+        let test_contract_balance = provider
+            .get_contract_asset_balance(test_contract._get_contract_id(), AssetId::default())
+            .await
+            .unwrap();
+        assert_eq!(test_contract_balance, 420);
+    }
+}
+
+// Test the cases where the transaction should panic due to the
+// predicate script failing to validate the transaction requirements
+mod panic {
+    use std::str::FromStr;
+
+    use crate::utils::builder;
+    use crate::utils::environment as env;
+    use fuels::prelude::Salt;
+    use fuels::prelude::TxParameters;
+    use fuels::test_helpers::DEFAULT_COIN_AMOUNT;
+    use fuels::tx::{Address, AssetId, Input, Output, Transaction, TxPointer, UtxoId, Word};
+
+    pub const RANDOM_SALT: &str =
+        "0xf55c10bc8307552781a896ebd5e6d2b9278b4177b8bca400d3e7710eee293786";
+
+    #[tokio::test]
+    #[should_panic(expected = "The transaction contains a predicate which failed to validate")]
+    async fn relay_message_with_too_many_inputs() {
+        let message_data = env::prefix_contract_id(vec![]).await;
+        let message = (100, message_data);
+        let coin1 = (1_000_000, AssetId::default());
+        let coin2 = (1_000_000, AssetId::default());
+        let coin3 = (1_000_000, AssetId::default());
+        let coin4 = (1_000_000, AssetId::default());
+        let coin5 = (1_000_000, AssetId::default());
+        let coin6 = (1_000_000, AssetId::default());
+        let coin7 = (1_000_000, AssetId::default());
+
+        // Set up the environment
+        let (wallet, _, contract_input, coin_inputs, message_inputs) = env::setup_environment(
+            vec![coin1, coin2, coin3, coin4, coin5, coin6, coin7],
+            vec![message],
+        )
+        .await;
+
+        // Build the message relaying transaction
+        let mut tx = builder::build_contract_message_tx(
+            message_inputs[0].clone(),
+            contract_input,
+            &coin_inputs[..],
+            &vec![],
+            &vec![],
             TxParameters::default(),
         )
-        .await
-        .unwrap();
+        .await;
 
-    // Check set up completed correctly
-    let predicate_balance = get_balance(&provider, predicate_root, native_asset).await;
-    assert_eq!(predicate_balance, transfer_amount);
+        // Sign transaction and call
+        // Note: tx inputs[contract, message, coin1, coin2, coin3, coin4, coin5, coin6, coin7], tx outputs[contract, change]
+        let _receipts = env::sign_and_call_tx(&wallet, &mut tx).await;
+    }
 
-    // Get the predicate coin to spend
-    let gas_coin = &provider.get_coins(&wallet.address()).await.unwrap()[0];
-    let predicate_coin = &provider.get_coins(&predicate_root).await.unwrap()[0];
+    #[tokio::test]
+    #[should_panic(expected = "The transaction contains a predicate which failed to validate")]
```

Comment:
Note that the `expected` is not currently supported so you cannot pass in anything and expect it to actually catch that specific reason

Reply:
I think it's supported. If this "expected" test is not in the error, the tests will fail. I actually ran into this while developing the tests, which helped me catch a flaw in my test setup. Maybe it works here because the error is at the predicate level and not a custom error thrown by a contract

Reply:
This has been an issue in the apps for a while so perhaps it's a feature that the SDK now supports but I haven't noticed that it was announced. I'll need to look into this.

Reply:
It seems to be supported.
My test passes with this :
```
#[tokio::test]
    #[should_panic(expected = "Revert(42)")]
```
but fails if I modify the expected to be `Revert(43)`. 
I'm not sure what sort of limitations exist on returned values that can be matched though.